File: aspdotnet/twitter-showcase/Models/Tweet.cs
using System.Dynamic;
using System.Text.Json;
using System.Text.Json.Serialization;

namespace twitter_showcase
{
    public class Tweet
    {
        [JsonPropertyName("id")]
        public int Id { get; set; }
        [JsonPropertyName("source")]
        public string Source { get; set; }
        [JsonPropertyName("user.profile_image_url_https")]
        public string ProfileImageUrl { get; set; }
        [JsonPropertyName("user.screen_name")]
        public string UserName { get; set; }
        [JsonPropertyName("user.name")]
        public string DisplayName { get; set; }
        [JsonPropertyName("created_at")]
        public string Date { get; set; }
        [JsonPropertyName("text")]
        public string Text { get; set; }
        [JsonPropertyName("favorite_count")]
        public int LikeCount { get; set; }
        [JsonPropertyName("retweet_count")]
        public int RetweetCount { get; set; }

        public override string ToString() => JsonSerializer.Serialize<Tweet>(this);
        
    }
}

File: API/client/src/Tweets/Tweet.js
import React from "react";
import TwitterThumbnail from "./../Images/twitter_thumbnail.png";
import tweetBg from "./../Images/tweet-bg.png";
import verifiedImage from "./../Images/verified.png";
import getSource from "../helpers/GetSource";
import Search from "../MainPageComponents/Search";

export default function Tweet({ tweetData, search }) {
    const verified = tweetData.user.verified ? (
        <img src={verifiedImage} alt="verified" style={{ height: 20, width: 20 }}></img>
    ) : (
        ""
    );
    console.log(tweetData);
    const source = getSource(tweetData.source);
    function handleSearch(name, type) {
        search(name, type);
    }

    return (
        <div className="tweet col-12" style={{ backgroundImage: `url(${tweetBg})` }}>
            <div className="row">
                <img
                    src={tweetData.user.profileImageUrlHttps}
                    style={{ paddingLeft: 10, borderRadius: "50%" }}
                    alt="thumbnail"
                ></img>

                <h4 className="col-lg-4">
                    {tweetData.user.name} {verified}
                </h4>
                <h5
                    className="col-lg-4 username"
                    style={{ cursor: "pointer" }}
                    onClick={() => search(tweetData.user.screenName, "user")}
                >
                    @{tweetData.user.screenName}
                </h5>
                <h6 className="col-lg-2">{tweetData.date.substring(4, 16)}</h6>

                <img
                    src={TwitterThumbnail}
                    style={{ maxWidth: 50, maxHeight: 50, paddingLeft: 10, paddingBottom: 10, float: "right" }}
                    alt="thumbnail"
                ></img>
            </div>
            <div className="tweet-text">
                <div dangerouslySetInnerHTML={{ __html: tweetData.formattedText }}></div>
            </div>

            <div>
                <div className="row">
                    <div className="row col-8">
                        <span role="img" aria-label="like" className="col-2">
                            💗 {tweetData.likeCount}
                        </span>
                        <span role="img" aria-label="retweet" className="col-2">
                            🔄 {tweetData.retweetCount}
                        </span>
                        <h6 className="username col-7" style={{ textAlign: "right" }}>
                            {source}
                        </h6>
                        <div className="col-1"></div>
                    </div>
                </div>
            </div>
        </div>
    );
}
